0xProject/0x-monorepo
# Improve 'web3-provider-engine' typings

## Description

Improvements for `web3-provider-engine` typings based on https://github.com/MetaMask/provider-engine/blob/master/index.js#L17

## Testing instructions

Before this pull, such source-code can not be compiled from typescript.
```typescript
import { Web3Wrapper } from '@0xproject/web3-wrapper';
import ProviderEngine from 'web3-provider-engine';
const engine = new ProviderEngine({ pollingInterval: 1000 });
// adding subproviders
export const instance = new Web3Wrapper(engine);

export async function start () {
 return new Promise(res => engine.start(res));
}

export async function stop () {
  engine.stop();
}
```
Now it can be compiled and work as expected.

## Types of changes


## Checklist:
